code/datums/components/nanites.dm

```diff
@@ -191,28 +285,28 @@
 	holder.icon_state = "nanites[nanite_percent]"
 
 /datum/component/nanites/proc/on_emp(datum/source, severity)
-	nanite_volume *= (rand(60, 90) * 0.01)		//Lose 10-40% of nanites
-	adjust_nanites(null, -(rand(5, 50)))		//Lose 5-50 flat nanite volume
-	if(prob(40/severity))
+	severity *= emp_severity_mod
+	var/loss = (severity / 100) * (rand(emp_percent_deletion_lower, emp_percent_deletion_upper) * nanite_volume) + rand(emp_flat_deletion_lower, emp_flat_deletion_upper)
+	adjust_nanites(null, -loss)
+	if(prob(severity / emp_desync_mod))
```

Comment:
I recommend doing this, since emp_desync_mod is < 1, therefore dividing through it would increase the probability, not decrease it.
```suggestion
	if(prob(severity * emp_desync_mod))
```